src/box/lua/iproto.c

```diff
@@ -268,16 +299,124 @@ push_iproto_protocol_features(struct lua_State *L)
 					  lengthof(features));
 }
 
+/**
+ * Encodes Lua table packet part to MsgPack using IPROTO key translation table:
+ * uses @a ibuf to store the result and saves the result length in @a mp_len.
+ */
+static const char *
+encode_lua_table_packet_part_to_mp(struct lua_State *L, int idx, size_t *mp_len)
+{
+	struct region *gc = &fiber()->gc;
+	struct mpstream stream;
+	mpstream_init(&stream, gc, region_reserve_cb, region_alloc_cb,
+		      luamp_error, L);
+	size_t used = region_used(gc);
+	luamp_encode_with_translation(L, luaL_msgpack_default, &stream, idx,
+				      iproto_key_translation);
+	mpstream_flush(&stream);
+	*mp_len = region_used(gc) - used;
+
+	const char *mp = region_join(gc, *mp_len);
+	if (mp == NULL) {
+		diag_set(OutOfMemory, *mp_len, "mp", "MsgPack data");
```

Comment:
Allocation from `fiber()->gc` should never fail. Let's use `xregion_join` here.

Reply:
Added `xregion_join` helper. Though the other code is written in this style. Could you please open a ticket regarding this matter?

Reply:
There's a lot of old code that uses `region_truncate`. We convert it gradually as we go, where applicable, and we write new code using x-allocators. There's no ticket for this, because this task isn't planned, nor tracked, and it doesn't make any difference to the user. I don't want to file one, since its scope is too wide.

Quoting @alyapunov: Filing a ticket for this is the same as filing a ticket to change long licenses to SPDX in all source files.

Reply:
My concern that not everyone agree with this memory management policy. But if there is a consensus on this vision, let it be.